### PR TITLE
Allow search by keyId and allow portal to use edit endpoints

### PR DESCRIPTION
**Pull request by joshua-smee-ttd**
Added a new endpoint to allow to search for client key by siteId. Added button to ui.
image

Also allowed portal to use edit endpoints to allow it to edit apiKeys.

### CODE REVIEW

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java [hunk 1]
```diff
@@ -208,6 +191,46 @@ private void handleClientListBySite(RoutingContext rc) {
...
+
+            rc.response()
+                    .putHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+                    .end(getClientReturnObject(clientKey).encode());
+        } catch (Exception e) {
+            rc.fail(500, e);
+        }
+    }
+
+    private static JsonObject getClientReturnObject(LegacyClientKey clientKey) {
```
Comment (alex-yau-ttd) [inline src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:220]: when we did this for `SiteService.java`, we called it `createSiteJsonObject`. Perhaps we could call this one `createClientKeyJsonObject` or similar?

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java [hunk 2]
```diff
@@ -208,6 +191,46 @@ private void handleClientListBySite(RoutingContext rc) {
         }
     }
 
+    private void handleClientByKeyId(RoutingContext rc) {
+        try {
+            String keyId;
+
+            try{
```
Comment (alex-yau-ttd) [inline src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:198]: I wonder if there is a more elegant approach to this that avoids having to nest `try-catch`es. What about something similar to the `clientKey` below like:
```
String keyId = rc.queryParam("keyId").stream().findFirst().orElse(null);

if (keyId == null) {
    ResponseUtil.error(rc, 400, "unable to parse keyId ");
    return;
}
```

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java [hunk 3]
```diff
@@ -208,6 +191,46 @@ private void handleClientListBySite(RoutingContext rc) {
...
+                keyId = rc.queryParam("keyId").get(0);
+            } catch (Exception e) {
+                ResponseUtil.error(rc, 400, "unable to parse keyId " + e.getMessage());
+                return;
+            }
+
+            LegacyClientKey clientKey = this.clientKeyProvider.getAll().stream().filter(legacyClientKey -> legacyClientKey.getKeyId().equals(keyId)).findFirst().orElse(null);
```
Comment (alex-yau-ttd) [inline src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:205]: I know it's only a one-liner, but it's a pretty long line. I wonder if it is worth extracting to:
```
private LegacyClientKey getClientKey(String keyId) {
    return this.clientKeyProvider.getAll()
            .stream()
            .filter(legacyClientKey -> legacyClientKey.getKeyId().equals(keyId))
            .findFirst()
            .orElse(null);
}
```

Comment (alex-yau-ttd) [reply src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:205]: Means we can better separate concerns - all I care about at this level of abstraction is that I'm getting a client key based off the `keyId`

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java (hunk 2 above)
Comment (joshua-smee-ttd) [reply src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:198]: Ah that's much nicer!

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java (hunk 3 above)
Comment (joshua-smee-ttd) [reply src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:205]: Yeah that makes sense!

**Review by alex-yau-ttd — APPROVED** [review]

**Pull request merged**